### PR TITLE
[Soar 17409] - Microsoft Teams - Adding fedramp true flag

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

https://rapid7.atlassian.net/browse/SOAR-17409
 - Adding the fedramp true flag
 - Adding in a option for the user to select what endpoints they want to connect to, normal, gcc, gcc high and MoD
 - Bumping the SDK to version 6.1.2
 - Bumping the version of zipp used to address a snyk issue 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

The unit tests have been updated to now include the resource url

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

Testing has been done locally with both actions and triggers and all works as expected

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by ekelly-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]

File: plugins/microsoft_teams/plugin.spec.yaml [hunk 1]
```diff
@@ -489,6 +540,17 @@ connection:
...
+    title: Endpoint
+    description: "The type of endpoint to connect to: normal service, GCC (government), GCC High (government), DoD (military)"
+    type: string
+    required: true
+    example: Normal
+    enum:
+    - Normal
+    - GCC
+    - "GCC High"
+    - DoD
```
Comment (igorski-r7) [inline plugins/microsoft_teams/plugin.spec.yaml:553]:
```suggestion
      - Normal
      - GCC
      - "GCC High"
      - DoD
```

File: plugins/microsoft_teams/icon_microsoft_teams/connection/connection.py [hunk 2]
```diff
@@ -17,10 +37,12 @@ def __init__(self):
         self.password = None
         self.api_token = None
         self.refresh_token = None
+        self.resource_endpoint = None
 
     def connect(self, params):
         self.app_id = params.get(Input.APPLICATION_ID)
         self.tenant_id = params.get(Input.DIRECTORY_ID)
+        self.endpoint = params.get(Input.ENDPOINT)
```
Comment (igorski-r7) [inline plugins/microsoft_teams/icon_microsoft_teams/connection/connection.py:45]:
```suggestion
        self.endpoint = params.get(Input.ENDPOINT, "Normal")
```

**Review by igorski-r7 — APPROVED** [review]
LGTM 😄 Left a small suggestions

File: plugins/microsoft_teams/plugin.spec.yaml (hunk 1 above)
Comment (rbowden-r7) [reply plugins/microsoft_teams/plugin.spec.yaml:553]: The indentation has been fixed

File: plugins/microsoft_teams/icon_microsoft_teams/connection/connection.py (hunk 2 above)
Comment (rbowden-r7) [reply plugins/microsoft_teams/icon_microsoft_teams/connection/connection.py]: The default has been added

**Pull request merged**